daemon/api_icons.go

```diff
@@ -69,3 +76,32 @@ func iconGet(st *state.State, name string) Response {
 
 	return fileResponse(icon)
 }
+
+func snapAppIconGet(c *Command, r *http.Request, user *auth.UserState) Response {
+	vars := muxVars(r)
+	snap := vars["snap"]
+	app := vars["app"]
+
+	snapInfo, err := localSnapInfo(c.d.overlord.State(), snap)
+	if err != nil {
+		if errors.Is(err, errNoSnap) {
+			return SnapNotFound(snap, err)
+		}
+		return InternalError(fmt.Sprintf("%v", err))
+	}
+
+	for _, appInfo := range snapInfo.info.Apps {
+		if appInfo.Name != app {
+			continue
+		}
+
+		iconPath, err := iconPathFromDesktopFile(appInfo.DesktopFile())
```

Comment:
I think that you don't need this to manage `.desktop` files. In `rest_api.go`, it is done using `goconfigparser`: https://github.com/snapcore/snapd/blob/4e85f27d0593e7d8c45bfd46f5279cfa38255056/usersession/agent/rest_api.go#L415

Reply:
Thanks!